### PR TITLE
make estimator config in helm values as a list

### LINKED ISSUE #4368: Support schedulerEstimator as a list in karmada helm chart
**What would you like to be added**:
- Support schedulerEstimator as a list in karmada helm chart
- add command line argument in [karmada-scheulder.yaml](https://github.com/karmada-io/karmada/blob/master/charts/karmada/templates/karmada-scheduler.yaml#L48-L53) to enable estimator like
  ```
  - --enable-scheduler-estimator=true
  ```
- Could we also make `"helm.sh/hook-delete-policy": hook-succeeded` configurable in [post-install-job.yaml](https://github.com/karmada-io/karmada/blob/master/charts/karmada/templates/post-install-job.yaml#L20C5-L20C33). We want to keep the job pod for debugging issues if post-install job fails

**Why is this needed**:
Hi, we want to install scheduler estimator using helm chart. But the helm char only supporting one cluster https://github.com/karmada-io/karmada/blob/master/charts/karmada/values.yaml#L748. I manually install one for another member cluster. Could we make them as a list ?

Thank you !

### PR DESCRIPTION
**Pull request by chaosi-zju**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

make estimator config in helm values as a list

https://github.com/karmada-io/karmada/issues/4050#issuecomment-1832454205

**Which issue(s) this PR fixes**:

Fixes #4368

**Special notes for your reviewer**:

image

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by chaosi-zju** [comment]
CC @RainbowMango ready to merge

File: charts/karmada/templates/karmada-scheduler-estimator.yaml [hunk 1]
```diff
@@ -106,22 +109,24 @@ stringData:
...
       app: karmada-scheduler-estimator-{{ $clusterName }}
-      {{- include "karmada.schedulerEstimator.labels" . | nindent 6 }}
-  {{ toYaml .Values.schedulerEstimator.podDisruptionBudget | nindent 2 }}
+      {{- include "karmada.schedulerEstimator.labels" $ | nindent 6 }}
+  {{ toYaml $.Values.schedulerEstimator.podDisruptionBudget | nindent 2 }}
 {{- end -}}
 
 {{- end }}
+
+{{- end }}
```
Comment (RainbowMango) [inline charts/karmada/templates/karmada-scheduler-estimator.yaml:132]: end of line

Comment (chaosi-zju) [reply charts/karmada/templates/karmada-scheduler-estimator.yaml]: done

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**